_pages/documentation/learning_gem5/part2/part2_1_helloobject.md

```diff
@@ -11,8 +11,11 @@ author: Jason Lowe-Power
 Creating a *very* simple SimObject
 ==================================
 
-**Note**: gem5 has SimObject named `SimpleObject`. Implementing another
-`SimpleObject` SimObject will result in confusing compiler issues.
+**Note**: You can find the files mentioned in this tutorial completed under
```

Comment:
I think we should keep the name in the tutorial and the name in the source code different. 

I don't want to encourage people to delete code. Instead, I think the best thing is for new users to create an object that has a different name from the source.

We put this in the source code mostly for testing so that this documentation doesn't break completely.

That said, let me know if I'm misunderstanding this change.

Reply:
The motivation for this PR was my confusion that the tutorial creates a simple `HelloObject`, yet the `HelloObject.py` and `hello_object.*` files in [`src/learning_gem5/part2`](https://github.com/gem5/gem5/tree/stable/src/learning_gem5/part2) - which are erroneously said to contain the complete code for this tutorial - are more sophisticated. Instead the files that match the `HelloObject` as described in this tutorial are called `SimpleObject` (`SimpleObject.py` and `simple_object.*`) in [`src/learning_gem5/part2`](https://github.com/gem5/gem5/tree/stable/src/learning_gem5/part2).
Whether this naming discrepancy is resolved by changing the tutorial or the files in [`src/learning_gem5/part2`](https://github.com/gem5/gem5/tree/stable/src/learning_gem5/part2) does not matter to me (I assumed it's preferred to change the documentation in [gem5/website](https://github.com/gem5/website) instead of the code in the main project [gem5/gem5](https://github.com/gem5/gem5)).


As for encouraging people to delete the code; this was merely a refactoring from later on in the tutorial on [lines 64-65](https://github.com/gem5/website/pull/80/files/8d49f2f7a1acb6bb8de19f426cdde749228645e2#diff-ddc78981f0d1ad5843702d973d1cd6ad30f3eba32c2add9b762617c4f77d615eL64-L65):
> You can delete these or move them elsewhere to follow this tutorial.

Reply:
The file `HelloObject.py` contains the code for creating a simple `HelloObject`. However, it also contains additional code because we add more code to the same file later on in the tutorial.

> You can delete these or move them elsewhere to follow this tutorial.
This comment intends to suggest that you can delete those files from your computer, if doing so would make it easier for you to follow the tutorial, rather than deleting them from the repository itself.

Please let me know if this clarifies your question.

Reply:
I understand that the deleting/moving/renaming part is for following the tutorial - it literally says "... to follow this tutorial. Of course locally, not in the remote repository. Again, this was just refactored from [later on in the tutorial](https://github.com/gem5/website/pull/80/files/8d49f2f7a1acb6bb8de19f426cdde749228645e2#diff-ddc78981f0d1ad5843702d973d1cd6ad30f3eba32c2add9b762617c4f77d615eL64-L65).